src/libraries/Microsoft.Bcl.Numerics/src/System/MathF.cs

```diff
@@ -0,0 +1,318 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+/*============================================================
+**
+** Purpose: Some single-precision floating-point math operations
+**
+===========================================================*/
+
+//This class contains only static members and doesn't require serialization.
+
+using System.Runtime.CompilerServices;
+using System.Diagnostics.Contracts;
+using System.Drawing;
+
+namespace System
+{
+    /// <summary>
+    /// Provides constants and static methods for trigonometric, logarithmic, and other common mathematical functions.
+    /// </summary>
+    public static class MathF
+    {
+        public const float PI = 3.14159265f;
+
+        public const float E = 2.71828183f;
+
+        private static float NegativeZero = Int32BitsToSingle(unchecked((int)0x80000000));
+
+        private static unsafe float Int32BitsToSingle(int value)
+        {
+            return *((float*)&value);
+        }
+
+        [Pure]
+        private static unsafe bool IsNegative(float f)
+        {
+            return (*(uint*)(&f) & 0x80000000) == 0x80000000;
```

Comment:
This is slightly cheaper as `SingleToInt32Bits(f) < 0`

Reply:
Thta doesn't exist in .net standard 2.0. Thats why I pulled it out that way.